helix-term/src/ui/editor.rs

```diff
@@ -697,6 +738,196 @@ impl EditorView {
+
+            if let Some(indicator) = node.indicator.as_deref() {
+                // set the indicator
+                surface.set_string(context_area.x, context_area.y, indicator, indicator_style);
+                continue;
+            }
+
+            let line_num_anchor = text.line_to_char(node.line_nr);
+
+            // get all highlights from the latest point
+            let highlights = Self::doc_syntax_highlights(doc, line_num_anchor, 1, theme);
+
+            let mut renderer = TextRenderer::new(
+                surface,
+                doc,
+                theme,
+                view.offset.horizontal_offset,
+                context_area,
+            );
+            let mut new_offset = view.offset;
+            new_offset.anchor = line_num_anchor;
+
+            render_text(
+                &mut renderer,
+                text,
+                new_offset,
+                &doc.text_format(context_area.width, Some(theme)),
+                doc_annotations,
+                highlights,
+                theme,
+                line_decoration,
+                translated_positions,
+            );
+
+            context_area.y += 1;
+        }
+    }
+
+    /// Calculates the sticky nodes
+    fn calculate_sticky_nodes(
+        nodes: &Option<Vec<StickyNode>>,
+        doc: &Document,
+        view: &View,
+        config: &helix_view::editor::Config,
+    ) -> Option<Vec<StickyNode>> {
+        let syntax = doc.syntax()?;
+        let tree = syntax.tree();
+        let text = doc.text().slice(..);
+        let viewport = view.inner_area(doc);
+        let cursor_line = doc.selection(view.id).primary().cursor(text);
+
+        // Use the cached nodes to determine the current topmost viewport
+        let anchor_line = text.char_to_line(view.offset.anchor);
+        let top_first_byte =
+            text.line_to_byte(anchor_line + nodes.as_deref().map_or(0, |v| v.len()));
+
+        let visual_cursor_pos = view
+            .screen_coords_at_pos(doc, text, cursor_line)
+            .unwrap_or_default()
+            .row as u16;
+
+        if visual_cursor_pos == 0 {
+            return None;
+        }
+
+        if let Some(nodes) = nodes {
+            if nodes.iter().any(|node| {
+                node.top_first_byte == top_first_byte && (visual_cursor_pos as usize) >= nodes.len()
+            }) {
+                return Some(nodes.to_vec());
+            }
+        }
+
+        let context_nodes = doc
+            .language_config()
+            .and_then(|lc| lc.sticky_context_nodes.as_ref());
+
+        let mut parent = tree
+            .root_node()
+            .descendant_for_byte_range(top_first_byte, top_first_byte)
+            .and_then(|n| n.parent());
+
+        // context is list of numbers of lines that should be rendered in the LSP context
+        let mut context: Vec<StickyNode> = Vec::new();
+
+        while let Some(node) = parent {
+            let line = text.char_to_line(node.start_byte());
```

Comment:
you're using a byte to find a char when a single char could be multiple byte, that's ASCII-only and probably wrong if there is multi-byte chars before the one you're looking for

Reply:
Yes in @erasin output there is `-  20:        0x10870149f - ropey::slice::RopeSlice::char_to_line::h83b3a9f53cafd361`, probably from here

Reply:
ohh i see. i should be using `byte_to_line` i assume?